Makefile

```diff
@@ -14,9 +14,12 @@ BUILD_DATE:=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 ALL_ARCHITECTURES=amd64 arm arm64 ppc64le s390x
 export DOCKER_CLI_EXPERIMENTAL=enabled
 
+# Tools versions
+# --------------
+GOLANGCI_VERSION:=1.50.1
+
 # Computed variables
 # ------------------
-HAS_GOLANGCI:=$(shell which golangci-lint)
```

Comment:
Why is modification needed here?
Is the original way inappropriate?

Reply:
Well, replacing `which golangci-lint` by `golangci-lint version --format=short` is not strictly required (I can revert if you prefer); I used it to make `make verify-lint` use a **fixed / determined** version of golangci-lint. Currently, if you have already locally an old version of `golangci-lint` on your path, `make golangci` will not update it, which means you will not use the same version of golangci-lint as CI. That can lead to discrepancies.

Furthermore:
- `HAS_GOLANGCI` was defined twice, with the same value (line 19 and line 222)
- line 226, `$(go env GOPATH)` does not work (it should be `$(shell go env GOPATH)`, or simply `$(GOPATH)`, because `GOPATH` variable is defined line 20.
- Thus, on my local environment (and I guess, for most developer environments), `make golangci` tries to install golangci-lint in `/bin`, which is denied without sudo privileges. In CI, installing in `/bin` works. However, in CI, `$(GOPATH)/bin` is not in binary path; that's why I replaced `golangci-lint` by `$(GOPATH)/bin/golangci-lint` in the other make rules.

Note: I used the same makefile rules when I added golangci-lint to prometheus-adapter in https://github.com/kubernetes-sigs/prometheus-adapter/pull/540.

Reply:
OK. I have no other questions. Maybe we need in-depth reviews from others
/cc @serathius @dgrisonnet